examples/nlp/lasertagger/lasertagger_main.py

```diff
@@ -0,0 +1,368 @@
+        default='bert-base-cased',
+        type=str,
+        help='Name of the pre-trained model',
+        choices=nemo_nlp.nm.trainables.get_pretrained_lm_models_list(),
+    )
+    parser_infer.add_argument(
+        '--bert_checkpoint', default='bert-base-cased', type=str, help='Path to the pre-trained model checkpoint'
+    )
+    parser_infer.add_argument(
+        "--model_config_file", default=None, type=str, help="Path to LaserTagger config file in json format"
+    )
+    parser_infer.add_argument("--vocab_file", default=None, help="Path to the vocab file.")
+    parser_infer.add_argument("--max_seq_length", default=128, type=int)
+
+    return parser.parse_args()
+
+
+def _calculate_steps(num_examples, batch_size, num_epochs, warmup_proportion=0):
+    """Calculates the number of steps.
+
+  Args:
+	num_examples: Number of examples in the dataset.
+	batch_size: Batch size.
+	num_epochs: How many times we should go through the dataset.
+	warmup_proportion: Proportion of warmup steps.
+
+  Returns:
+	Tuple (number of steps, number of warmup steps).
+  """
+    steps = int(num_examples / batch_size * num_epochs)
+    warmup_steps = int(warmup_proportion * steps)
+    return steps, warmup_steps
+
+
+if __name__ == "__main__":
+
+    args = parse_args()
+
+    if args.command == 'train':
+        train_examples, num_train_examples = torch.load(args.train_file)
+        if args.eval_file:
+            eval_examples, num_eval_examples, eval_special_tokens = torch.load(args.eval_file)
+    test_examples, num_test_examples, test_special_tokens = torch.load(args.test_file_preprocessed)
+
+    label_map = utils.read_label_map(args.label_map_file)
+    num_tags = len(label_map)
+
+    config = None
+    with open(args.model_config_file, "r", encoding="utf-8") as reader:
+        text = reader.read()
+        config = json.loads(text)
+
+    nf = nemo_core.NeuralModuleFactory(
+        backend=nemo_core.Backend.PyTorch,
+        local_rank=args.local_rank,
+        optimization_level=args.amp_opt_level,
+        log_dir=args.work_dir,
+        create_tb_writer=True,
+        files_to_copy=[__file__],
+        add_time_to_log_dir=False,
+    )
+
+    encoder = nemo_nlp.nm.trainables.huggingface.BERT(pretrained_model_name=args.pretrained_model_name)
+
+    tokenizer = nemo_nlp.data.NemoBertTokenizer(pretrained_model=args.pretrained_model_name)
+
+    tokenizer.add_special_tokens({"additional_special_tokens": test_special_tokens})
+    # encoder.resize_token_embeddings(len(tokenizer))
+
+    vocab_size = 8 * math.ceil(tokenizer.vocab_size / 8)
+    tokens_to_add = vocab_size - tokenizer.vocab_size
+
+    encoder = nemo_nlp.nm.trainables.huggingface.BERT(pretrained_model_name=args.pretrained_model_name)
+
+    device = encoder.bert.embeddings.word_embeddings.weight.get_device()
+    zeros = torch.zeros((tokens_to_add, config['decoder_hidden_size'])).to(device=device)
+    encoder.bert.embeddings.word_embeddings.weight.data = torch.cat(
```

Comment:
why embedding weights are set to 0?

Reply:
This is extending the existing embeddings by calculating the number of tokens to add with zero weights. The original weights still remain.